### ISSUE TITLE
[Other - Accessibility] Delete button removes upload without user confirmation. (00.00.2)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#59263]
Needs a design ticket

#### VFS team name
Benefits Disability Experience Team

#### VFS product name
Disability Comp Application (21-526 EZ), Add Self Assessment Part A form information to BDD workflow

#### VFS feature name
Add Self Assessment Part A form information to BDD workflow

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
No VA.gov Experience Standard for the issue found.

### Issue Details
On [/supporting-evidence/additional-evidence/](https://staging.va.gov/disability/file-disability-claim-form-21-526ez/supporting-evidence/additional-evidence), when deleting an uploaded file, the Delete button is a single-click action with no step to confirm the action. Additionally, there's nothing announced by screen readers to indicate that a change has occurred --- focus just moves to the next focusable element.

It's easy to imagine some scenarios where a user might accidentally trigger the Delete button and not realize that they did it. Screen reader users may hit the wrong key but think they hit [Tab]. A low-vision user on a touch device might pinch-to-zoom and accidentally tap the Delete button in the process, and not see the issue disappear.

Since users get an opportunity to review everything prior to submission, this is probably not a violation of experience standard 03.09 or WCAG 3.3.4. But I do think covered by the spirit of WCAG 3.3.4.

### VA.gov Experience Standard
[Category Number 00, Issue Number 00](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Platform Recommendation
Two recommended changes:

1. Some kind of status message that receives focus when an issue has been removed, similar to the background-only alerts in use when changing contact information. If a visible status message isn't practical, at least some `aria-live` update should be announced to screen reader users.
2. Some kind of intermediate confirmation step between a user clicking Delete and the issue actually being removed. I believe the team working on the Supplemental Claims Form 20-0995 is exploring using a modal and intends to contribute back to the pattern in the design system.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by lydiahooper** [comment]
@RakshindaAslam @christinec-fftc We can discuss but I expect this will require some design work

**Comment by RakshindaAslam** [comment]
Will create a design ticket for this @RakshindaAslam

**Comment by lydiahooper** [comment]
@RakshindaAslam when you create the design ticket, please include the following info:
Ensure interactions work well for users with more than one document - usually select-save/upload interaction versus upload, label and then do nothing else pattern. Also consider UX for users who decide not to upload.

**Comment by lydiahooper** [comment]
Created design ticket to address: https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/63414

**Comment by mengA6** [comment]
confirmed in the design ticket that [there is a confirmation dialog already in place on staging](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/63414#issuecomment-1984455572).

**Comment by mengA6** [comment]
### Implementation ticket
[Add alert message after success or failure of a file deletion#78072](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/78072)
